src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -2753,29 +2753,34 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
     /** String representation of symbol's definition following its name */
     final def infoString(tp: Type): String = {
       def parents = (
-        if (settings.debug.value) parentsString(tp.parents)
+        if (settings.debug) parentsString(tp.parents)
         else briefParentsString(tp.parents)
       )
       def isStructuralThisType = (
         // prevents disasters like scala/bug#8158
         owner.isInitialized && owner.isStructuralRefinement && tp == owner.tpe
       )
-      if (isType) typeParamsString(tp) + (
-        if (isClass) " extends " + parents
-        else if (isAliasType) " = " + tp.resultType
-        else tp.resultType match {
-          case rt @ TypeBounds(_, _) => "" + rt
-          case rt                    => " <: " + rt
+      def loop(tp: Type, first: Boolean = false): String = {
+        // colon space, preceded by an extra space if needed to prevent the colon glomming onto a symbolic name
+        def postnominalColon: String = if (first && name.isOperatorName) " : " else ": "
+        if (isType) typeParamsString(tp) + (
+          if (isClass) " extends " + parents
+          else if (isAliasType) " = " + tp.resultType
+          else tp.resultType match {
+            case rt@TypeBounds(_, _) => "" + rt
+            case rt => " <: " + rt
+          }
+        )
+        else if (isModule) "" //  avoid "object X of type X.type"
+        else tp match {
+          case PolyType(tparams, res) => typeParamsString(tp) + loop(res)
```

Comment:
My last thought before the vp asked me to look at something was that this will trigger `-Xlint:recurse-with-default` when merged forward.

Reply:
Am I making life interesting for @SethTisue or whoever does the forward merges, or am I just being irritating? AFAIR we don't use that flag so maybe it's a pending enhancement for your next tranche of warnings fixups.

Reply:
That's the flag you approved in January on 2.13.